library/core/src/panicking.rs

```diff
@@ -72,6 +72,7 @@ pub const fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
 #[cfg_attr(feature = "panic_immediate_abort", inline)]
 #[cfg_attr(not(bootstrap), rustc_nounwind)]
 #[cfg_attr(bootstrap, rustc_allocator_nounwind)]
+#[track_caller]
```

Comment:
the doc comment explicitly says that this dliberately does not have track_caller.

Reply:
Hm, nice catch. I do wonder how much this matters compared to the unwinding size though. I'm not sure it's worth having a separate version either. It's definitely worth fixing the comment though, and possibly worth doing a perf run.

Reply:
I don't think we have a way to do perf runs for a stdlib with these debug assertions.

Reply:
And we definitely need a separate version to get the `rustc_nounwind` (so there shouldnt be any 'unwinding size').

Reply:
> And we definitely need a separate version to get the rustc_nounwind (so there shouldn't be any 'unwinding size').

This was my point...

Reply:
> I don't think we have a way to do perf runs for a stdlib with these debug assertions.

No, but it would be worth doing a perf run if we're going to enable `#[track_caller]` to see if turning this on matters for anything else. If it's only used in the preconditions, then I think it's fine to enable track_caller, if it's used elsewhere too... well, then it may depend some.

Reply:
This is only used in the debug assertions and the "abort because we are unwinding out of a `nounwind` function". The latter made a big point about not using track_caller and I'd like to stick with this for now here, too. I don't think those line numbers are worth it, since they are stdlib-internal.

I'd rather see the 'stringify' replaced by a hand-written string that would then also much easier let us see where the error comes from.

Reply:
> I don't think we have a way to do perf runs for a stdlib with these debug assertions.

I can certainly do some locally, if there's a subset of the scenarios you think would be most informative. When I was working on CTFE I did a lot of check-full runs.

Reply:
> We want this function to just have a single register-sized argument for minimal code size impact.

Not to be a pedant, but this function has a two-register argument. `rustc-perf` running only check-full reports this improvement if I change the argument to `&&'static str`: 
![2022-10-16-181236_799x388_scrot](https://user-images.githubusercontent.com/12105168/196060801-eeee9d40-2def-4f26-b5a5-a9b878f87469.png)

Reply:
I am confused, the argument is already a reference?

I was referring to `track_caller` which adds another argument, and means every call site needs to set up suitable location data.

Reply:
> &str is 16 bytes, passing the pointer and the length requires two registers. So currently, calls to this function are mov + mov + call

Ah, good point, I hadn't even realized this.

No strong opinion on whether it's worth saving that `mov`.

Reply:
I have a slightly more diabolical idea for making these calls cheaper, which I will pursue in a separate PR.